src/setup_node_env/node_version_validator.js

```diff
@@ -37,15 +37,18 @@ var pkg = require('../../package.json');
 var currentVersion = (process && process.version) || null;
 var rawRequiredVersion = (pkg && pkg.engines && pkg.engines.node) || null;
 var requiredVersion = rawRequiredVersion ? 'v' + rawRequiredVersion : rawRequiredVersion;
-var isVersionValid = !!currentVersion && !!requiredVersion && currentVersion === requiredVersion;
+var requiredVersionMajorMinor = requiredVersion.match(/^v(\d+\.\d+)/)[1];
+var isVersionValid = requiredVersionMajorMinor === currentVersion.match(/^v(\d+\.\d+)/)[1];
 
 // Validates current the NodeJS version compatibility when OpenSearch Dashboards starts.
 if (!isVersionValid) {
   var errorMessage =
```

Comment:
Can we add at least one test to confirm that this error message spits out what we expect? I don't see any current tests that validate these regex expressions.

Reply:
Added in a28c8ed248... but please verify that this is what you had in mind. I can't really test a fixed string because it changes with the used node version.